src/main/java/eu/xenit/alfred/content/gateway/runtime/config/ComposedApplicationConfiguration.java

```diff
@@ -0,0 +1,59 @@
+package eu.xenit.alfred.content.gateway.runtime.config;
+
+import eu.xenit.alfred.content.gateway.runtime.ApplicationId;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ComposedApplicationConfiguration implements ApplicationConfiguration {
+
+    @Getter
+    @NonNull
+    private final ApplicationId applicationId;
+
+    @Override
+    public String getConfigurationId() {
+        return this.getApplicationId().toString();
+    }
+
+    private final Map<String, ApplicationConfigurationFragment> fragments = new HashMap<>();
```

Comment:
I think it can be `ApplicationConfiguration` instead of explicitly `ApplicationConfigurationFragment`?